File: YoutubeClone/DTO/CommentSummary.cs
namespace YoutubeClone.DTO
{
    public class CommentSummary
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public Guid VideoId { get; set; }
        public Guid? ParentCommentId { get; set; }
        public string UserName { get; set; } = string.Empty;
        public string UserProfilePictureUrl { get; set; } = string.Empty;
        public string Text { get; set; } = string.Empty;
        public long Likes { get; set; }
        public long Dislikes { get; set; }
        public long Replies { get; set; }
        public DateTime DateCreated { get; set; }
    }
}

File: YoutubeClone/ClientApp/src/api/models/comment-summary.ts
export interface CommentSummary {
    id: string;
    userId: string;
    userName: string;
    userProfilePictureUrl: string;
    text: string;
    likes: number;
    dislikes: number;
    replies: number;
    dateCreated: Date;
}

export function mapJsonToCommentSummary(json: any): CommentSummary {
    return {
        id: json.id,
        userId: json.userId,
        userName: json.userName,
        userProfilePictureUrl: json.userProfilePictureUrl,
        text: json.text,
        likes: json.likes,
        dislikes: json.dislikes,
        replies: json.replies,
        dateCreated: new Date(json.dateCreated)
    }
}